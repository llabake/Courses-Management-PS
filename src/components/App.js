import React, { Component } from 'react';
import { connect } from "react-redux";

import PropTypes from 'prop-types';
import Header from "./common/Header";

class App extends Component {
  render() {
    return (
      <div className="container-fluid">
        <Header loading={this.props.loading}/>
        {this.props.children}
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired,
};

export const mapStateToProps = state => {
  return {
    loading: state.ajaxCallsInProgress > 0,
  };
};

// export default App;
export default connect(mapStateToProps)(App);
