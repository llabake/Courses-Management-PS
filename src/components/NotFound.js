import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
  <div>
    <div className="jumbotron">
      <h3 className="text-center">This is embarrassing, something is missing......</h3>
      <p className="text-center"><Link to="/">Home</Link></p>
    </div>
  </div>
);

export default NotFound;
