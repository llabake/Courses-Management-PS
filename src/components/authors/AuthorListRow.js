/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const AuthorListRow = ({ author, onDelete }) => {
  const deleteAuthor = () => (
    onDelete(author.id)
  );
  return (
    <tr>
      <td><Link to={`/author/${author.id}`}>{author.id}</Link></td>
      <td>
        {author.firstName} 
        {' '}
        {author.lastName}
      </td>
      <td
        className="btn btn-xs btn-danger"
        onClick={deleteAuthor}
      >
        Delete
      </td>
    </tr>
  );
};

AuthorListRow.propTypes = {
  author: PropTypes.object.isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default AuthorListRow;
