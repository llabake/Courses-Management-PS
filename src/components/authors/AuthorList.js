import React from 'react';
import PropTypes from 'prop-types';
import AuthorListRow from "./AuthorListRow";

const AuthorList = ({ authors, onDelete }) => (
  <table className="table">
    <thead>
      <tr>
        <th>ID</th>
        <th>Name</th>
        <th />
      </tr>
    </thead>
    <tbody>
      {authors.map(author => (
        <AuthorListRow
          key={author.id}
          author={author}
          onDelete={onDelete} />
      ))}
    </tbody>
  </table>
);

AuthorList.propTypes = {
  authors: PropTypes.array.isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default AuthorList;
