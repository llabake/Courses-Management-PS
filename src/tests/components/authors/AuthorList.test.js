import React from 'react';
import { shallow } from 'enzyme';
import AuthorList from "../../../components/authors/AuthorList";
import AuthorListRow from "../../../components/authors/AuthorListRow";
import authorMock from "../../mocks/authorMock";
import { mapStateToProps } from "../../../components/App";

describe('<AuthorList />', () => {
  it('renders correctly', () => {
    const props = {
      onDelete: jest.fn(),
      authors: authorMock.authors,
    };
    const wrapper = shallow(<AuthorList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders authorList Row correctly ', () => {
    const props = {
      onDelete: jest.fn(),
      authors: authorMock.authors,
    };
    const wrapper = shallow(<AuthorList {...props} />);
    expect(wrapper.find(AuthorListRow).length).toBe(2);
  });
});
