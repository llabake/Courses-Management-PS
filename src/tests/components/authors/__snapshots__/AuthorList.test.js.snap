// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorList /> renders correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        ID
      </th>
      <th>
        Name
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <AuthorListRow
      author={
        Object {
          "firstName": "author_one",
          "id": "mock-author-one",
          "lastName": "mock-author",
        }
      }
      key="mock-author-one"
      onDelete={[MockFunction]}
    />
    <AuthorListRow
      author={
        Object {
          "firstName": "author_two",
          "id": "cory-house",
          "lastName": "mock-author-two",
        }
      }
      key="cory-house"
      onDelete={[MockFunction]}
    />
  </tbody>
</table>
`;
